src/plugins/undoRedo/undoRedo.js

```diff
@@ -29,14 +29,21 @@ function UndoRedo(instance) {
   this.ignoreNewActions = false;
   this.enabled = false;
 
+  instance.addHook('beforeUndoStackChange', (action, filterBySource) => {
+    if (filterBySource === 'UndoRedo.undo' || filterBySource === 'UndoRedo.redo' || filterBySource === 'auto') {
+      return false;
+    }
+  });
```

Comment:
The `filterBySource` name gives me a wrong feeling that under the argument there's a `boolean`. May I suggest to change the argument name to the `source`? 
```suggestion
  instance.addHook('beforeUndoStackChange', (action, source) => {
    if (source === 'UndoRedo.undo' || source === 'UndoRedo.redo' || source === 'auto') {
      return false;
    }
  });
```